slack/web/urllib_client.py

```diff
@@ -0,0 +1,245 @@
+import copy
+import io
+import json
+import logging
+import mimetypes
+import uuid
+from http.client import HTTPResponse
+from typing import BinaryIO, Dict, List, Union
+from urllib.error import HTTPError
+from urllib.parse import urlencode
+from urllib.request import Request, urlopen
+
+from slack.web import get_user_agent
+from slack.web.slack_response import SlackResponse
+
+
+class HttpErrorResponse(object):
+    pass
+
+
+class UrllibWebClient:
+    logger = logging.getLogger(__name__)
+
+    def __init__(
+        self,
+        *,
+        token: str = None,
+        default_headers: Dict[str, str] = dict(),
+        # Not type here to avoid ImportError: cannot import name 'WebClient' from partially initialized module
+        # 'slack.web.client' (most likely due to a circular import
+        web_client=None,
+    ):
+        """urllib-based API client.
+
+        :param token: Slack API Token (either bot token or user token)
+        :param default_headers: request headers to add to all requests
+        :param web_client: WebClient instance for pagination
+        """
+        self.token = token
+        self.default_headers = default_headers
+        self.web_client = web_client
+
+    def api_call(
```

Comment:
It's also possible to use this method for any API calls. As described in slack_response.py, pagination iterator doesn't work when directly using this class. To use the feature, developers should use `WebClient` with `run_async=False`. The reason I gave up supporting the interaction with this class is circular import issues with `BaseClient`.